tests/Unit/ControlSystem/Test_PiecewisePolynomial.cpp

```diff
@@ -0,0 +1,120 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "ControlSystem/PiecewisePolynomial.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+SPECTRE_TEST_CASE("Unit.ControlSystem.FunctionsOfTime.PiecewisePolynomial",
+                  "[ControlSystem][Unit]") {
+  double t = 0.0;
+  const double dt = 0.6;
+  const double final_time = 4.0;
+  const constexpr size_t deriv_order = 3;
+  const constexpr size_t ret_deriv_order = 1;
+
+  // test two component system (x**3 and x**2)
+  const std::array<DataVector, deriv_order + 1> init_func{
+      {{0.0, 0.0}, {0.0, 0.0}, {0.0, 2.0}, {6.0, 0.0}}};
+  FunctionsOfTime::PiecewisePolynomial<deriv_order> f_of_t(t, init_func);
+
+  while (t < final_time) {
+    const auto& lambdas0 = f_of_t(t);
+    CHECK(approx(lambdas0[0][0]) == cube(t));
```

Comment:
you should be checking all the derivatives in this case

Reply:
Is there not a third derivative to check? (It may be trivial, but the function should still do the right thing.)